cea/technologies/thermal_network/thermal_network_costs.py

```diff
@@ -0,0 +1,563 @@
+
+    if network_info.config.thermal_network.network_type == 'DH':
+        deltaPmax = np.max(network_info.network_features.DeltaP_DHN)
+    else:
+        deltaPmax = np.max(network_info.network_features.DeltaP_DCN)
+    Capex_a, Opex_fixed = pumps.calc_Cinv_pump(deltaPmax, mdotnMax_kgpers, PUMP_ETA, network_info.config,
+                                               network_info.locator, 'PU1')  # investment of Machinery
+    Opex_a_tot = Opex_var + Opex_fixed
+
+    return Capex_a, Opex_a_tot
+
+
+def calc_Ctot_cooling_plants(network_info):
+    """
+    Calculates costs of centralized cooling plants (chillers and cooling towers).
+    :param network_info: an object storing information of the current network
+    :return:
+    """
+
+    # read in plant heat requirement
+    plant_heat_hourly_kWh = pd.read_csv(
+        network_info.locator.get_optimization_network_layout_plant_heat_requirement_file(
+            network_info.network_type, network_info.config.thermal_network_optimization.network_name))
+    # read in number of plants
+    number_of_plants = len(plant_heat_hourly_kWh.columns)
+
+    plant_heat_original_kWh = plant_heat_hourly_kWh.copy()
+    plant_heat_peak_kW_list = plant_heat_hourly_kWh.abs().max(axis=0).values  # calculate peak demand
+    plant_heat_sum_kWh_list = plant_heat_hourly_kWh.abs().sum().values  # calculate aggregated demand
+
+    Opex_var_chiller = 0.0
+    Opex_var_CT = 0.0
+    Capex_a_chiller = 0.0
+    Opex_a_chiller = 0.0
+    Capex_a_CT = 0.0
+    Opex_a_CT = 0.0
+    # calculate cost of chiller heat production and chiller capex and opex
+    for plant_number in range(number_of_plants):  # iterate through all plants
+        if number_of_plants > 1:
+            plant_heat_peak_kW = plant_heat_peak_kW_list[plant_number]
+        else:
+            plant_heat_peak_kW = plant_heat_peak_kW_list[0]
+        plant_heat_yearly_kWh = plant_heat_sum_kWh_list[plant_number]
+        print 'Annual plant heat production:', round(plant_heat_yearly_kWh, 0), '[kWh]'
+        Capex_chiller = 0
+        Opex_fixed_chiller = 0
+        Capex_CT = 0
+        Opex_fixed_CT = 0
+        if plant_heat_peak_kW > 0:  # we have non 0 demand
+            peak_demand_W = plant_heat_peak_kW * 1000  # convert to W
+            print 'Calculating cost of heat production at plant number: ', plant_number
+            if network_info.config.thermal_network_optimization.yearly_cost_calculations:
+                # calculates operation costs with yearly approximation
+                COP_plant = VCCModel.calc_VCC_COP(network_info.config,
+                                                  network_info.full_cooling_systems,
+                                                  centralized=True)
+                Opex_var_chiller += abs(
+                    plant_heat_yearly_kWh) / COP_plant * 1000 * network_info.prices.ELEC_PRICE
+                Opex_var_CT += CTModel.calc_CT_yearly(plant_heat_yearly_kWh)
```

Comment:
This model is problematic:
1. the heat input should be the heat rejected from the condenser, not the cooling production from plants
2. there is no bounds for this equation

Reply:
1. Ok I see the problem, but without a model of the plant I'm not sure how to solve this, except maybe by trying to find a factor which relates plant heat to condenser heat? Some kind of plant condenser efficiency?

2. The CT function has an upper bound in the sense that it will just add more cooling towers, lower bound should be 0. But I think I am misunderstanding what you mean :) 

Please advise on how to proceed on these two points :)

Reply:
It seems like this is actually no longer relevant since the CT OPEX is included in the OPEX var chiller calculation.

Reply:
Problably this is redudant as Opex_var_CT seems to be included in the Opex_cooling_plant calculation

Reply:
@shanshanhsieh 
This seems to be the only point which is still open - the CT Capex is still being calculated based on sizing of the CT to the heat demand, not to the heat rejected from the condenser. Any ideas how to fix this?